packages/files-ui/src/Components/Modules/FileBrowsers/CreateShareModal.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useRef, useEffect, useState, useCallback } from "react"
+import CustomModal from "../../Elements/CustomModal"
+import { CSFTheme } from "../../../Themes/types"
+import { useFiles } from "../../../Contexts/FilesContext"
+import { useFilesApi } from "../../../Contexts/FilesApiContext"
+import CustomButton from "../../Elements/CustomButton"
+import { t, Trans } from "@lingui/macro"
+import { LookupUser, LookupUserRequest } from "@chainsafe/files-api-client"
+import EthCrypto from "eth-crypto"
+import clsx from "clsx"
+
+const useStyles = makeStyles(
+  ({ breakpoints, constants, typography, zIndex, palette }: CSFTheme) => {
+    return createStyles({
+      root: {
+        padding: constants.generalUnit * 4,
+        flexDirection: "column",
+        display: "flex",
+        alignItems: "center"
+      },
+      modalRoot: {
+        zIndex: zIndex?.blocker,
+        [breakpoints.down("md")]: {}
+      },
+      modalInner: {
+        backgroundColor: constants.createShareModal.backgroundColor,
+        color: constants.createShareModal.color,
+        [breakpoints.down("md")]: {
+          bottom:
+          Number(constants?.mobileButtonHeight) + constants.generalUnit,
+          borderTopLeftRadius: `${constants.generalUnit * 1.5}px`,
+          borderTopRightRadius: `${constants.generalUnit * 1.5}px`,
+          maxWidth: `${breakpoints.width("md")}px !important`
+        }
+      },
+      okButton: {
+        marginLeft: constants.generalUnit
+      },
+      cancelButton: {
+        [breakpoints.down("md")]: {
+          position: "fixed",
+          bottom: 0,
+          left: 0,
+          width: "100%",
+          height: constants?.mobileButtonHeight
+        }
+      },
+      label: {
+        fontSize: 14,
+        lineHeight: "22px"
+      },
+      heading: {
+        color: constants.createShareModal.color,
+        fontWeight: typography.fontWeight.semibold,
+        marginBottom: 10
+      },
+      iconBacking: {
+        backgroundColor: constants.createShareModal.iconBackingColor,
+        width: 48,
+        height: 48,
+        borderRadius: 24,
+        marginBottom: 16,
+        "& > svg": {
+          width: 16,
+          height: 16,
+          fill: palette.primary.main,
+          position: "relative",
+          display: "block",
+          transform: "translate(-50%, -50%)",
+          top: "50%",
+          left: "50%"
+        }
+      },
+      modalFlexItem: {
+        width: "100%",
+        margin: 5
+      },
+      buttons: {
+        justifyContent: "flex-end",
+        display: "flex"
+      }
+    })
+  }
+)
+
+interface ICreateShareModalProps {
+  modalOpen: boolean
+  close: () => void
+}
+
+const CreateShareModal: React.FC<ICreateShareModalProps> = ({
+  modalOpen,
+  close
+}: ICreateShareModalProps) => {
+  const classes = useStyles()
+  const { createShare } = useFiles()
+  const { filesApiClient } = useFilesApi()
+  const [creatingShare, setCreatingShare] = useState(false)
```

Comment:
For boolean state `is*` would be great
```suggestion
  const [isCreatingSharedFolder, setIsCreatingSharedFolder] = useState(false)
```